catalog/dags/providers/provider_api_scripts/auckland_museum.py

```diff
@@ -0,0 +1,143 @@
+
+Resource:               https://api.aucklandmuseum.com/
+                        https://github.com/AucklandMuseum/API/wiki/Tutorial
+"""
+import logging
+
+from common.constants import IMAGE
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+logger = logging.getLogger(__name__)
+
+LANDING_URL = (
+    "https://www.aucklandmuseum.com/collections-research/collections/record/am_"
+)
+
+
+class AucklandMuseumDataIngester(ProviderDataIngester):
+    providers = {
+        "image": prov.AUCKLAND_MUSEUM_IMAGE_PROVIDER,
+    }
+    endpoint = "https://api.aucklandmuseum.com/search/collectionsonline/_search"
+    license_url = "https://creativecommons.org/licenses/by/4.0/"
+    delay = 4
+    from_start = 0
+    total_amount_of_data = 10000
+    DEFAULT_LICENSE_INFO = get_license_info(license_url=license_url)
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # On the first request, `prev_query_params` will be `None`. We can detect this
+        # and return our default params.
+        if not prev_query_params:
+            # Return default query params on the first request
+            # primaryRepresentation contain a image url for each data
+            # "+" is a query string syntax for must be present
+            # copyright:CC state Creative Commons Attribution 4.0
+            return {
+                "q": "_exists_:primaryRepresentation+copyright:CC",
+                "size": "100",
+                "from": self.from_start,
+            }
+        else:
+            # Increment `from` by 100.
+            return {
+                **prev_query_params,
+                "from": prev_query_params["from"] + 100,
+            }
+
+    def get_batch_data(self, response_json):
+        # Takes the raw API response from calling `get` on the endpoint, and returns
+        # the list of records to process.
+        if response_json:
+            return response_json.get("hits").get("hits")
+        return None
+
+    def get_should_continue(self, response_json):
+        # Do not continue if we have exceeded the total amount of data
+        if self.from_start >= self.total_amount_of_data:
+            logger.info(
+                "The final amount of data has been processed. Halting ingestion."
+            )
+            return False
+
+        return True
+
+    def get_media_type(self, record: dict):
+        return IMAGE
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        url_parameter = data.get("_id").split("id/")[-1].replace("/", "-")
+        foreign_landing_url = f"{LANDING_URL}{url_parameter}"
+
+        foreign_identifier = data.get("_id").split("/")[-1]
+
+        information = data.get("_source")
+
+        url = information.get("primaryRepresentation")
+
+        thumbnail_url = f"{url}?rendering=thumbnail.jpg"
```

Comment:
The thumbnail their API provides is tiny, fixed width of 70px. @obulat would know best -- should we use this or just default to `None` here and use our own thumbnail service? They also have a slightly bigger `preview` rendering with a fixed width of 100px.

Reply:
We have previously discussed the thumbnail sizes, and decided against using thumbnails smaller than 600px: https://github.com/WordPress/openverse/issues/675#issuecomment-1472399310